app/src/main/java/com/futurumgame/base/gameinternals/WareHouse.java

```diff
@@ -64,7 +80,102 @@ public boolean canOfferResources(LinkedList<Resource> requiredResources) {
 
     public void offerResources(LinkedList<Resource> requiredResources) {
         for (Resource requiredResource : requiredResources) {
-            ((Resource) wareHouseStocks.get(requiredResource.getID())).subtract(requiredResource.getCount());
+            Resource resource = wareHouseStocks.get(requiredResource.getID());
+            resource.subtract(requiredResource.getCount());
+        }
+    }
+
+    @Override
+    public void from(byte[] data) {
+        if (data.length == 0) {
```

Comment:
What happens when data is null?

Reply:
data is never null due to the use of the method

Reply:
Okay. But see above.